sbt-app/src/sbt-test/actions/cross-strict-aggregation-scala-3/build.sbt

```diff
@@ -0,0 +1,14 @@
+scalaVersion := "2.12.16"
+
+lazy val core = project
+  .settings(
+    crossScalaVersions := Seq("2.12.16", "3.0.2", "3.1.2")
+  )
+
+lazy val subproj = project
+  .dependsOn(core)
+  .settings(
+    crossScalaVersions := Seq("2.12.16", "3.1.2"),
+    // a random library compiled against Scala 3.1
+    libraryDependencies += "org.http4s" %% "http4s-core" % "0.23.12"
```

Comment:
I know depending on a 3rd party is probably not a good thing to do in SBT tests. Do you have any idea on how I can force compilation to fail if we use Scala 3.0 (in other words, ensure that this project only compiles with Scala 3.1)?

Reply:
Maybe you could define a custom task or macro within the project that's aware of the Scala version of the compiler?